behave_framework/src/minifi_test_framework/steps/checking_steps.py

```diff
@@ -84,6 +84,13 @@ def step_impl(context: MinifiTestContext, message: str, duration: str):
                               context=context)
 
 
+@then("the Minifi logs contain the following message: \"{log_message}\" {count:d} times after {duration}")
+def step_impl(context, log_message, count, duration):
+    duration_seconds = humanfriendly.parse_timespan(duration)
+    time.sleep(duration_seconds)
+    assert context.get_default_minifi_container().get_logs().count(log_message) == count or context.get_default_minifi_container().log_app_output()
```

Comment:
What does the second half mean? `.log_app_output()`
It seems like we ignore the assertion if it's true.